### ISSUE TITLE
Make gantt view to show also retries

### ISSUE THREAD

**Issue by alexInhert** [apache/airflow#17487]
**Description**

Currently Gantt View shows only the last successful task retry.
This means that if we have t1 >> t2 >> t3
if we had:
t1 started on 08:00 finished 09:00
t2  1st retry started 09:01 failed on 13:00
t2 2nd retry started on 13:02 finished on 15:00
t3 started on 15:01 finished on 15:02

What the gantt view will show us is 
t1 08:00-09:00
t2 13:02-15:00
t3 15:01-15:02

This makes it really hard to understand what really happen because it doesn't show what really happened.
This is very annoying for DAGs with so many tasks because you can't tell if the problem is due to tasks not being scheduled or this is something else. I understand the value of the current display but maybe it can add some toggle to consider also all the task instances?

**Use case / motivation**

The gantt somehow should show the actual information.

**Are you willing to submit a PR?**

Don't have the skills for that.

**Comment by bbovenzi** [comment]
I've been thinking about how to make the gantt view more useful (I think it has a lot of potential). Showing the full state history of a task instance would be a great part of that.

**Comment by ShakaibKhan** [comment]
I would like to try implementing this

**Comment by bbovenzi** [comment]
Sweet. Just assigned you. Feel free to reach out if you need anything!

**Comment by alexInhert** [comment]
@ShakaibKhan @bbovenzi  see this example
![Screen Shot 2021-11-30 at 20 27 28](https://user-images.githubusercontent.com/62940116/144106608-e5ca04ba-6c36-429c-b6f4-2610de9f7780.png)


As you can see there is significant gap between the tasks. When you hover over the task you see it is the 2nd retry. If going into the log it shown that the first retry run for the duration of the purple rectangle.
My request is to add the purple rectangle. I want to see also the retries in the gantt without it - the Gantt just looks like the scheduler didn't schedule anything during this time which is incorrect view of what really happened.
I think the current setup is useful I'm asking for a way to enrich it also with the retries of the tasks to show the full picture.

**Comment by ShakaibKhan** [comment]
@alexInhert make sense to me and agree that retries should be shown; could you please share the DAG you used to make the above gantt chart

**Comment by ShakaibKhan** [comment]
Sorry I do not think I can take on this issue at this time. Unassigning to myself

**Comment by yxiao1996** [comment]
Reproduce the scenario described in this issue with a python operator which causes failed task at 50% chance

```
@task(task_id=f"sleep_and_randomly_fail", retries=1)
def my_sleeping_function():
    time.sleep(5)
    if random.uniform(0, 1) > 0.5:
        raise ValueError("Mock Exception.")
```

From the DAG run results it seems to me this issue is already addressed, as in the following snapshot we can see both failed 1st run and successful 2nd run in gantt chart. Under the hood, reruns of a tasks are stored as different task instances. 
![Screenshot_20230108_112349](https://user-images.githubusercontent.com/27112484/211214947-15feed15-fda2-41f4-a0be-51bcd70ff5b2.png)

![Screenshot_20230108_112408](https://user-images.githubusercontent.com/27112484/211214952-57f1e6ec-adc2-422d-a4f1-a85ced688a9f.png)

**Comment by bbovenzi** [comment]
Yes, we record Task Fails which are included in the Gantt chart. We also record Task Reschedules which would probably be helpful. Renaming this issue.

**Comment by bbovenzi** [comment]
Any other improvements will come with [AIP-64](https://cwiki.apache.org/confluence/display/AIRFLOW/AIP-64%3A+Keep+TaskInstance+try+history)

**Comment by jscheffl** [comment]
Implemented in #40304 and #40357 - will be part of Airflow 2.10!
